### PR TITLE
Add findNonZero

### PR DESCRIPTION
**Pull request by ocharles**

### CODE REVIEW

File: opencv/src/OpenCV/Core/ArrayOps.hs [hunk 1]
```diff
@@ -1058,3 +1059,28 @@ perspectiveTransform srcPoints transformationMat = unsafePerformIO $
...
     c'numPts = fromIntegral $ V.length srcPoints
+
+-- | Find all non-zero points of a matrix.
+findNonZero
+    :: Mat shape ('S 1) depth
+    -> V.Vector Point2i
+findNonZero mat = unsafePerformIO $
+  withPtr mat $ \srcPtr ->
+  alloca $ \(numPointsPtr :: Ptr Int32) ->
+  alloca $ \(pointsPtrPtr :: Ptr (Ptr (Ptr C'Point2i))) -> do
```
Comment (basvandijk) [inline opencv/src/OpenCV/Core/ArrayOps.hs]: We need a `mask_` here to guarantee the finalization of the points array.

Comment (ocharles) [reply opencv/src/OpenCV/Core/ArrayOps.hs]: Good catch. Just `mask_` the whole thing?

Comment (basvandijk) [reply opencv/src/OpenCV/Core/ArrayOps.hs]: Yes, a `mask_` before the `do` should do the right thing.

**Comment by ocharles** [comment]
Ok, added a `mask_`.

**Comment by basvandijk** [comment]
Thanks.

**Pull request merged**